lib/collection/query-param.js

```diff
@@ -136,26 +136,34 @@ _.assign(QueryParam, /** @lends QueryParam */ {
      *
      * @param {Array|Object} params
      * @param {Object=} options
-     * @param {Boolean} options.encode - Enables URL encoding of the parameters
+     * @param {?Boolean} [options.encode=false] - Enables URL encoding of the parameters
+     * @param {?Boolean} [options.ignoreDisabled=false] - Removes disabled query parameters when set to true.
      * @returns {string}
      */
     unparse: function (params, options) {
-        var encode = options && options.encode,
-            str;
-
         if (!params) { return EMPTY; }
 
+        // Convert hash maps to an array of params
         if (!_.isArray(params) && !PropertyList.isPropertyList(params)) {
             params = _.transform(params, function (accumulator, value, key) {
                 accumulator.push({ key: key, value: value });
             }, []);
         }
 
-        str = params.map(function (param) {
-            return QueryParam.unparseSingle(param, encode);
-        }).join(AMPERSAND);
+        var encode = options && options.encode,
+            ignoreDisabled = options && options.ignoreDisabled,
+
+            // construct a query parameter string from the list, with considerations for disabled values
+            str = params.reduce(function (result, param) {
```

Comment:
Keep str; in var and move the function body out of var definition. One level less indentation and that the main work of this function is then not tucked inside a var.